### PR TITLE
Document multi-user installation, and add release notes about it not being the default on 2.1

### PR DESCRIPTION
**Pull request by grahamc**
Rendered docs:

image

Rendered release notes:

image

Ideally would be backported.

### CODE REVIEW

File: doc/manual/installation/installing-binary.xml [hunk 1]
```diff
@@ -6,13 +6,119 @@
...
+    </listitem>
+    <listitem><para>macOS</para></listitem>
+  </itemizedlist>
+
+  <para>
+    You can force the installer to attempt a multi-user installation on
+    your system, even if it is unsupported:
+
+    <screen>
+  curl https://nixos.org/nix/install | sh -s -- --daemon
```
Comment (ElvishJerricco) [inline doc/manual/installation/installing-binary.xml]: Won't this give

```
No TTY, assuming you would say yes :)
```

regarding the y/n questions in the multi-user installer? That seems like a bad thing to recommend by default.

Comment (grahamc) [reply doc/manual/installation/installing-binary.xml]: You're right!

File: doc/manual/installation/installing-binary.xml [hunk 2]
```diff
@@ -6,13 +6,119 @@
...
-$ bash &lt;(curl https://nixos.org/nix/install)
+  $ bash &lt;(curl https://nixos.org/nix/install)
 </screen>
 
+If your system is supported by the multi-user installer, the
+installation tool will default to installing a multi-user Nix daemon.
+</para>
+
+<para>
+I you would like to explicitly opt in or out of the multi-user Nix
```
Comment (LnL7) [inline doc/manual/installation/installing-binary.xml]: typo

File: doc/manual/installation/installing-binary.xml [hunk 3]
```diff
@@ -6,13 +6,119 @@
...
+# If you are on Linux with systemd, you will need to run:
+sudo systemctl stop nix-daemon.socket
+sudo systemctl stop nix-daemon.service
+sudo systemctl disable nix-daemon.socket
+sudo systemctl disable nix-daemon.service
+sudo systemctl daemon-reload
+
+# If you are on macOS, you will need to run:
+sudo launchctl unload /Library/LaunchDaemons/org.nixos.nix-daemon.plist
+sudo launchctl unload /Library/LaunchDaemons/org.nixos.nix-daemon.plist
```
Comment (LnL7) [inline doc/manual/installation/installing-binary.xml]: just once should be enough 😄

File: doc/manual/installation/installing-binary.xml [hunk 4]
```diff
@@ -6,13 +6,119 @@
...
+<section xml:id="sect-single-user-installation">
+  <title>Single User Installation</title>
+
+  <para>
+    You can force the installer to perform a single-user installation
+    on your system, even if it is supported by the multi-user
+    installer:
+
+    <screen>
+  curl https://nixos.org/nix/install | sh -s -- --no-daemon
```
Comment (LnL7) [inline doc/manual/installation/installing-binary.xml]: Does passing flags to `bash <(command)` not work?

File: doc/manual/installation/installing-binary.xml (hunk 1 above)
Comment (grahamc) [reply doc/manual/installation/installing-binary.xml]: Fixed by your recommendation on IRC, thank you!

File: doc/manual/installation/installing-binary.xml (hunk 4 above)
Comment (grahamc) [reply doc/manual/installation/installing-binary.xml]: Fixed!

File: doc/manual/installation/installing-binary.xml (hunk 3 above)
Comment (grahamc) [reply doc/manual/installation/installing-binary.xml]: Fixed!

File: doc/manual/installation/installing-binary.xml (hunk 2 above)
Comment (grahamc) [reply doc/manual/installation/installing-binary.xml]: Fixed!

**Comment by grahamc** [comment]
Rebased and squashed commits.

File: doc/manual/installation/installing-binary.xml [hunk 5]
```diff
@@ -6,13 +6,118 @@
...
+sudo systemctl disable nix-daemon.socket
+sudo systemctl disable nix-daemon.service
+sudo systemctl daemon-reload
+
+# If you are on macOS, you will need to run:
+sudo launchctl unload /Library/LaunchDaemons/org.nixos.nix-daemon.plist
+sudo rm /Library/LaunchDaemons/org.nixos.nix-daemon.plist
+</screen>
+
+    There may also be references to Nix in
```
Comment (LnL7) [inline doc/manual/installation/installing-binary.xml]: I can't remember, are these steps skipped during install if references to /nix are found in these files? Or does it fail.

Comment (grahamc) [reply doc/manual/installation/installing-binary.xml]: It fails.

**Comment by grahamc** [comment]
Updated rendering: https://screenshotscdn.firefoxusercontent.com/images/7cb7afcb-bc81-4cb2-a680-2a7c2ba83d02.png

**Comment by grahamc** [comment]
Running an install test suite.

**Comment by grahamc** [comment]
Install matrix says things are good (full report attached):

note macos failed because I only built a tarball for linux. The changed part of the PR worked fine:

> macos-sierra-install    Note: a multi-user installation is possible. See https://nixos.org/nix/manual/#sect-multi-user-installation
> 
[report-single-user-default.html.gz](https://github.com/NixOS/nix/files/2342411/report-single-user-default.html.gz)
macos-sierra-install    performing a single-user installation of Nix...
  
image

File: doc/manual/release-notes/rl-2.1.xml [hunk 6]
```diff
@@ -49,8 +49,38 @@ new features:</para>
...
+  </listitem>
+
 </itemizedlist>
 
+<para>
+  Multi-user Nix users on macOS can upgrade Nix by running
+  <command>sudo -i sh -c 'nix-channel --update &amp;&amp; nix-env -iA
+  nixpkgs.nix'; sudo launchctl stop org.nixos.nix-daemon; sudo
+  launchctl start org.nixos.nix-daemon</command>.
+</para>
```
Comment (edolstra) [inline doc/manual/release-notes/rl-2.1.xml]: I think the release notes are not the best place for platform-specific upgrade instructions, which are presumably the same for every release. Might be better to refer to some "Upgrading Nix" section.

Comment (grahamc) [reply doc/manual/release-notes/rl-2.1.xml]: Fixed!

**Comment by tomberek** [comment]
Recommend adding a link to 4.4 to where a user can find the latest Nix binary tarball.

**Comment by grahamc** [comment]
What is the title of 4.4 you'd like a link to?

**Pull request merged**